include/eve/traits/overload/protocol.hpp

```diff
@@ -0,0 +1,158 @@
+-> decltype(std::declval<TYPE>().call(std::declval<Args>()...))                                                        \
+requires( requires { std::declval<TYPE>().call(EVE_FWD(args)...); })                                                   \
+{                                                                                                                      \
+  if constexpr(eve::decorated_callable<TYPE,Args...>)                                                                  \
+    return TYPE::deferred_call(eve::current_api, eve::detail::defaults<TYPE,Args...>(), EVE_FWD(args)...);             \
+  else                                                                                                                 \
+    return TYPE::deferred_call(eve::current_api, EVE_FWD(args)...);                                                    \
+}                                                                                                                      \
+static EVE_FORCEINLINE decltype(auto) deferred_call(auto arch, auto&&...args) noexcept                                 \
+requires(requires { NAME(eve::detail::adl_helper, arch, EVE_FWD(args)...); })                                          \
+{                                                                                                                      \
+  return NAME(eve::detail::adl_helper, arch, EVE_FWD(args)...);                                                        \
+}                                                                                                                      \
+template<typename... Args>                                                                                             \
+static EVE_FORCEINLINE decltype(auto) deferred_call(auto arch, Args&&...args) noexcept                                 \
+requires(!requires { NAME(eve::detail::adl_helper, arch, EVE_FWD(args)...); })                                         \
+{                                                                                                                      \
+  return eve::detail::default_behavior<TYPE,Args...>::call(arch, EVE_FWD(args)...);                                    \
+}                                                                                                                      \
+using callable_tag_type     = TYPE                                                                                     \
+/**/
+
+//======================================================================================================================
+//! @addtogroup extensions
+//! @{
+//!   @def EVE_CALLABLE_NAMESPACE
+//!   @brief Register a namespace as suitable for containing eve::callable overloads.
+//! @}
+//======================================================================================================================
+#define EVE_CALLABLE_NAMESPACE()                                                                                       \
```

Comment:
Why function like macro?

Reply:
It looks better in IDE and prevent syntax hightlighting fuck up